modules/openapi-generator/src/main/resources/dart2/serialization/native/native_class.mustache

```diff
@@ -86,7 +86,7 @@ class {{{classname}}} {
     {{/isDate}}
     {{^isDateTime}}
       {{^isDate}}
-      json[r'{{{baseName}}}'] = this.{{{name}}};
+      json[r'{{{baseName}}}'] = this.{{{name}}}{{#isArray}}{{#uniqueItems}}{{#isNullable}}!{{/isNullable}}.toList(growable: false){{/uniqueItems}}{{/isArray}};
```

Comment:
```suggestion
      json[r'{{{baseName}}}'] = this.{{{name}}}{{#isArray}}{{#uniqueItems}}{{#isNullable}}?{{/isNullable}}.toList(growable: false){{/uniqueItems}}{{/isArray}};
```

shouldn't using `?` be better than `!` ?

Reply:
Few things:
1) The generated code includes a null check one line before: `if (json['field'] != null) …`.
2) In other parts of the template, a bang `!` is used instead of a `?`, so just following suit.

Reply:
alright then looks good to me

Reply:
we definitely need to add test schemas to check the changes before they are published 
see my comment here https://github.com/OpenAPITools/openapi-generator/pull/13047#issuecomment-1536754543

Reply:
Agreed. If I understood correctly from your comment, are you writing the tests already?

Reply:
yes, I am writing the tests in a separate pr

Reply:
related https://github.com/OpenAPITools/openapi-generator/issues/15449